apps/antalmanac/src/components/RightPane/CoursePane/CourseRenderPane.tsx

```diff
@@ -162,134 +180,87 @@ export function CourseRenderPane() {
             division: formData.division,
         };
 
-        const websocQueryParams = {
-            department: formData.deptValue,
-            term: formData.term,
-            ge: formData.ge,
-            courseNumber: formData.courseNumber,
-            sectionCodes: formData.sectionCode,
-            instructorName: formData.instructor,
-            units: formData.units,
-            endTime: formData.endTime,
-            startTime: formData.startTime,
-            fullCourses: formData.coursesFull,
-            building: formData.building,
-            room: formData.room,
-            division: formData.division,
-        };
-
-        const gradesQueryParams = {
-            department: formData.deptValue,
-            ge: formData.ge as GE,
-        };
-
         try {
-            // Query websoc for course information and populate gradescache
-            const [websocJsonResp, _] = await Promise.all([
-                websocQueryParams.units.includes(',')
-                    ? queryWebsocMultiple(websocQueryParams, 'units')
-                    : queryWebsoc(websocQueryParams),
-                Grades.populateGradesCache(gradesQueryParams),
-            ]);
-
+            let jsonResp;
+            if (params.units.includes(',')) {
+                jsonResp = await queryWebsocMultiple(params, 'units');
+            } else {
+                jsonResp = await queryWebsoc(params);
+            }
+            setLoading(false);
             setError(false);
-            setCourseData(flattenSOCObject(websocJsonResp));
+            setWebsocResp(jsonResp);
+            setCourseData(flattenSOCObject(jsonResp));
         } catch (error) {
-            setError(true);
-        } finally {
             setLoading(false);
```

Comment:
Why did you remove the `finally` clause? Does the order of `setLoading`, `setWebsocResp`, and `setCourseData` matter?